geofence/geofence.go

```diff
@@ -0,0 +1,45 @@
+package echogeofence
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+// IsAddressNearChecker is an interface that guarantees the middleware is compatible with multiple libraries
+// Currently satisfied with the go-geofence library
+type IsAddressNearChecker interface {
+	IsIPAddressNear(ipAddress string) (bool, error)
+}
+
+// Middleware looks up the coordinates of a client to see if it's nearby
+func Middleware(geofence IsAddressNearChecker) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			// Get client ip
+			ipAddress := c.RealIP()
+
+			// Ensure not a private ip
+			ip := net.ParseIP(ipAddress)
+			if ip.IsPrivate() || strings.HasPrefix(ipAddress, "::1") {
```

Comment:
maybe conceptionally it would even better if those IP checks are moved into geofence library and done inside `IsIPAddressNear` method. Reasoning is that these check are needed for everyone and it would be more ergonomic if user does not need to implement those parts.

Reply:
That's fair. I've added it to the library, removed here, and updated the example usage doc

Reply:
Any additional thoughts on the changes?

Reply:
@aldas any other concerns here?